### PR TITLE
hdf5 hdf5-mpi 1.14.5

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```diff
- I/O filters (external):  DEFLATE DECODE ENCODE
+ I/O filters (external):  DEFLATE
```

**Comment by cho-m** [comment]
Breakage caused by https://github.com/HDFGroup/hdf5/pull/4567

I think upstream is aware based on https://github.com/HDFGroup/hdf5/issues/4949#issuecomment-2408286244

File: Formula/h/hdf5.rb [hunk 1]
```diff
@@ -124,3 +121,37 @@ def install
...
+diff --git a/CMakeFilters.cmake b/CMakeFilters.cmake
+index 52d65e5..b43aa3b 100644
+--- a/CMakeFilters.cmake
++++ b/CMakeFilters.cmake
+@@ -108,7 +108,7 @@ if (HDF5_ENABLE_Z_LIB_SUPPORT)
+         # on the target. The target returned is: ZLIB::ZLIB
+         get_filename_component (libname ${ZLIB_LIBRARIES} NAME_WLE)
+         string (REGEX REPLACE "^lib" "" libname ${libname})
+-        set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME zlib-static)
++        set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME z)
```
Comment (cho-m) [inline Formula/h/hdf5.rb:135]: Upstream fix would probably be
```suggestion
+        set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME ${libname})
```

Comment (cho-m) [reply Formula/h/hdf5.rb:135]: Upstreamed this one. Other patch is taking a while upstream.

**Review by daeho-ro — APPROVED** [review]

**Pull request merged**